### PR TITLE
Fix LatchPublisher peer_subscribe to publish all key-values

### PR DESCRIPTION
**Pull request by leandropineda**
# Changes

Fix `LatchPublisher` publish method to avoid overriding `message` property when publishing messages. Also extended `peer_subscribe` to publish all key-values when a peer subscribes.

See https://docs.ros.org/en/melodic/api/rospy/html/rospy.topics.SubscribeListener-class.html

## How to reproduce

Republisher configuration file

```
republishers:
- topic: "/foo"
  msg_type: "sensor_msgs/Temperature"
  latched: true
  mappings:
  - field: "temperature"
    mapping_type: "single_field"
    out:
      topic: "/inorbit/custom_data/0"
      key: "temperature"
  - field: "variance"
    mapping_type: "single_field"
    out:
      topic: "/inorbit/custom_data/0"
      key: "variance"
```

Use the following script to publish a single latched message and keep the node running forever

```
#!/usr/bin/env python

import rospy
from sensor_msgs.msg import Temperature

def talker():
    pub = rospy.Publisher('/foo', Temperature, queue_size=10, latch=True)
    rospy.init_node('talker', anonymous=True)

    msg = Temperature(temperature=1.0, variance=2.0)
    rospy.loginfo(msg)
    pub.publish(msg)

    rate = rospy.Rate(10) # 10hz
    while not rospy.is_shutdown():
        rate.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass
```

Subscribe to the `/inorbit/custom_data/0` topic and start the republisher `roslaunch inorbit_republisher example.launch`. Only the last `key` will be published.

```
data: "variance=2.0"                             
---                                              
data: "variance=2.0"                             
---                                              
data: "variance=2.0"                             
---                                              
data: "variance=2.0"                             
```

### CODE REVIEW

File: inorbit_republisher/launch/example.launch [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <launch>
-  <node name="inorbit_republisher" pkg="inorbit_republisher" type="republisher.py">
+  <node name="inorbit_republisher" pkg="inorbit_republisher" type="republisher.py" output="screen">
```
Comment (FlorGrosso) [inline inorbit_republisher/launch/example.launch]: @leandropineda is this a leftover from your tests?

File: inorbit_republisher/scripts/republisher.py [hunk 2]
```diff
@@ -298,15 +297,19 @@ def serialize(msg, mapping):
 class LatchPublisher(rospy.Publisher, rospy.SubscribeListener):
     def __init__(self, name, data_class, tcp_nodelay=False, headers=None, queue_size=None):
         super(LatchPublisher, self).__init__(name, data_class=data_class, tcp_nodelay=tcp_nodelay, headers=headers, queue_size=queue_size, subscriber_listener=self, latch=False)
-        self.message = None
+        self.latch_publisher = super(LatchPublisher, self)
+        # Store all kv messages when publising.
```
Comment (FlorGrosso) [inline inorbit_republisher/scripts/republisher.py]:
```suggestion
        # Store all kv messages when publishing.
```

File: inorbit_republisher/launch/example.launch (hunk 1 above)
Comment (leandropineda) [reply inorbit_republisher/launch/example.launch]: No, I left it on purpose so messages printed to stdout are shown.

Comment (FlorGrosso) [reply inorbit_republisher/launch/example.launch]: Mmm I see, the only flag I have about this is how verbose the republisher becomes. I believe this is the reason why no ouput to screen was set.

**Review by FlorGrosso — APPROVED** [review]
Thanks for the update, Lean! LGTM

File: inorbit_republisher/launch/example.launch (hunk 1 above)
Comment (leandropineda) [reply inorbit_republisher/launch/example.launch]: I agree, I'll revert it JIC then

**Pull request merged**